### PR TITLE
Ensure we quote the cte table name if needed

### PR DESCRIPTION
**Pull request by pgammans**
Postgresql an possibly other dbms requires the CTE name to be quoted unless it is lower case unquoted CTE names are always folded to lower case. This causes CTE that use a camel case name to break if referenced from a subquery or annotation.

This patch ensure we use the django's active compiler to escaping the CTE name.

### CODE REVIEW

File: django_cte/raw.py [hunk 1]
```diff
@@ -19,15 +19,17 @@ def get_source_expressions(self):
             return []
 
     class raw_cte_compiler(object):
-        @staticmethod
-        def as_sql():
+        def as_sql(self, ):
```
Comment (millerdev) [inline django_cte/raw.py:22]: Why was this changed? I don't think it's necessary, but if it is please change to `def as_sql(self):`

File: django_cte/raw.py [hunk 2]
```diff
@@ -19,15 +19,17 @@ def get_source_expressions(self):
...
             return sql, params
 
+        def quote_name_unless_alias(self, name):
+            return name
```
Comment (millerdev) [inline django_cte/raw.py:26]: Should this use `qn(name)`? Could also make this a `@staticmethod` and undo the change on line 32 below since `self` is not referenced.

**Review by millerdev — COMMENTED** [review]
Is it possible to test the changes made to `raw_cte_sql`?

File: django_cte/raw.py (hunk 2 above)
Comment (pgammans) [reply django_cte/raw.py:26]: It actually is the value of qn used in CTECompiler.generate_sql if the query is a raw query.
for non raw queries this function comes from the base django.db.models.sql.compiler.SQLCompiler which use the connection to provide the correct escaping function. I was just ignoring escaping raw queries. 

I have now added a test to check that raw queries to escape the CTE name.
A static would have worked, but fixing escaping the raw cte name requires it again as we need to store the connection.

**Review by millerdev — APPROVED** [review]
Thanks for the contribution.
